sdcm/tester.py

```diff
@@ -249,13 +160,13 @@ def __init__(self, methodName='test', name=None, params=None,
         cluster.Setup.reuse_cluster(self.params.get('reuse_cluster', default=False))
 
         # for saving test details in DB
-        self.test_index = self.__class__.__name__.lower()
-        self.test_type = self.params.id.name
-        self.test_id = None
-        self.stats = {k: {} for k in ['test_details', 'setup_details', 'versions', 'results', 'nemesis',
-                                      'errors', 'coredumps']}
+        self._db_stats = db_stats.TestStats(self)
         self.create_stats = True
 
+    @property
```

Comment:
Why do you use @property here?
Property decorator is used to replace "setters" and "getters" for conveniency of implementing "data encapsulation principle"
I think you shouldn't use it since it is not a property of this class and is set on __init__

Reply:
Nemesis has to access the TestStats object for adding info to test statistics - so it passed while creating nemesis object in test (see add_nemesis()).